### DISCUSSION TITLE
Rolling Restart?

### DISCUSSION THREAD

**Discussion by theturkeyneck** [kubernetes-client/csharp#868] [Q&A]
Hello, 
I am pretty new to c# and kb8s and I was wondering if I could perform a rolling restart using this package.
I wanted to try it running locally but I had some issues running the examples.
Let me add that I cannot modify any metadata and fluxCD will prevent changes to the deployment.

Thank you for your help!
Jorge

**Comment by brendandburns** [comment]
This package is mostly just a C# library for interacting with the Kubernetes API. So anything you can do in the API you can do with this library.

I'm not sure what you mean by "rolling restart" but if you just want to terminate a bunch of Pods so that the ReplicaSet controller will restart them, then yes, you can do that, effectively by calling (pseudo-code)

```
for each pod in client.listPods(...) {
    client.deletePod(pod)
}
```

**Comment by theturkeyneck** [comment]
hi, thank you for your answer!
by "rolling restart" I mean that at least one pod must be always up, so whenever we are doing any updates to it, it will need to scale up one pod and load the updates while the other one is still working and after it is done loading, terminate the older one, so the APIs consuming it never stop and fail (that's why we have 2 pods always running just in case one fails)

**Comment by brendandburns** [comment]
Oh, in that case all you need to do is update the image in the `V1Deployment` object.

There's a patch example here:

https://github.com/kubernetes-client/csharp/tree/master/examples/patch

It shows how to patch a V1Pod, but you could adapt it to patch a V1Deployment. If you change the image for the Deployment from `foo:v1` to `foo:v2` then the Deployment logic in Kubernetes will do the right thing.

See the docs here:
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

for more details on how deployments work.
